internal/connector/internal/api/dbapi/processor.go

```diff
@@ -0,0 +1,117 @@
+package dbapi
+
+import (
+	"github.com/bf2fc6cc711aee1a0c2a/kas-fleet-manager/pkg/api"
+	"github.com/bf2fc6cc711aee1a0c2a/kas-fleet-manager/pkg/db"
+)
+
+type ProcessorDesiredState string
+type ProcessorStatusPhase string
+
+const (
+	ProcessorReady   ProcessorDesiredState = "ready"
+	ProcessorStopped ProcessorDesiredState = "stopped"
+	ProcessorDeleted ProcessorDesiredState = "deleted"
+
+	ProcessorStatusPhasePreparing      ProcessorStatusPhase = "preparing"      // set by kas-fleet-manager - user request
+	ProcessorStatusPhasePrepared       ProcessorStatusPhase = "prepared"       // set by kas-fleet-manager - worker
+	ProcessorStatusPhaseUpdating       ProcessorStatusPhase = "updating"       // set by kas-fleet-manager - user request
+	ProcessorStatusPhaseStopped        ProcessorStatusPhase = "stopped"        // set by kas-fleet-manager - user request
+	ProcessorStatusPhaseProvisioning   ProcessorStatusPhase = "provisioning"   // set by kas-agent
+	ProcessorStatusPhaseReady          ProcessorStatusPhase = "ready"          // set by the agent
+	ProcessorStatusPhaseFailed         ProcessorStatusPhase = "failed"         // set by the agent
+	ProcessorStatusPhaseDeprovisioning ProcessorStatusPhase = "deprovisioning" // set by kas-agent
+	ProcessorStatusPhaseDeleting       ProcessorStatusPhase = "deleting"       // set by the kas-fleet-manager - user request
+	ProcessorStatusPhaseDeleted        ProcessorStatusPhase = "deleted"        // set by the agent
+)
+
+var ValidProcessorDesiredStates = []string{
+	string(ProcessorReady),
+	string(ProcessorStopped),
+	string(ProcessorDeleted),
+}
+
+var AgentProcessorStatusPhase = []string{
+	string(ProcessorStatusPhaseProvisioning),
+	string(ProcessorStatusPhaseDeprovisioning),
+	string(ProcessorStatusPhaseStopped),
+	string(ProcessorStatusPhaseReady),
+	string(ProcessorStatusPhaseFailed),
+	string(ProcessorStatusPhaseDeleted),
+}
+
+type Processor struct {
+	db.Model
+	Name            string
+	NamespaceId     string
+	ProcessorTypeId string
+	Channel         string
```

Comment:
@manstis Apologies if we have already discussed this, but do you know what is the status of support for `channel` on Connector and if it's something that is being actively used?

Reply:
Support for different Connector `channel`'s was added recently [here](https://github.com/bf2fc6cc711aee1a0c2a/kas-fleet-manager/commit/6e3c3d817dc4b4dab4a95937b877722bc2c637ae). I don't know how actively used it is. We could choose to remove this field for `Processor`'s (for now) but I suspect it may follow.

Reply:
yes they are used now, primary as a gate for filtering out connectors depending on the maturity level so i.e. come connectors won't be available in prod but they will be in dev/staging. Same thinking can be applied to preview conenctors at some point